src/Aspire.Dashboard/Components/Controls/PropertyGrid.razor.cs

```diff
@@ -6,10 +6,62 @@
 
 namespace Aspire.Dashboard.Components.Controls;
 
-public partial class PropertyGrid<TItem>
+/// <summary>
+/// Describes an name/value item to be displayed in a <see cref="PropertyGrid{TItem}"/>.
+/// </summary>
+/// <remarks>
+/// <para>
+/// Implement this interface to use as the <c>TItem</c> of a <see cref="PropertyGrid{TItem}"/> component.
+/// </para>
+/// <para>
+/// The property grid has two columns, bound to display strings <see cref="Name"/> and <see cref="Value"/>.
+/// </para>
+/// <para>
+/// The <see cref="IsValueSensitive"/> and <see cref="IsValueMasked"/> properties control masking behavior,
+/// which prevents sensitive data from being displayed in the UI without user interaction.
+/// </para>
+/// </remarks>
+public interface IPropertyGridItem
```

Comment:
nit: Call this IPropertyGridViewModel

The standard throughout the UI is to suffix view model to types that are designed to be databound.

Reply:
Thanks but I'll leave it as is. It's an interface which describes a role rather than a concrete type. `PropertyGrid` refers to its items as "item", so this keeps things consistent.